app/panel/components/PlusPromoModal.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * PlusPromoModal Component
+ *
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2019 Ghostery, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClassNames from 'classnames';
+import Modal from '../../shared-components/Modal';
+import ModalExitButton from './BuildingBlocks/ModalExitButton';
+
+/**
+ * A functional React component for a PlusPromo Modal that may be displayed in the Hub and/or Panel
+ * @return {JSX} JSX for rendering a PlusPromo Modal
+ * @memberof SharedComponents
+ */
+const PlusPromoModal = (props) => {
```

Comment:
Is there a way we could combine `PlusPromoModal`, `InsightsPromoModal`, and `PremiumPromoModal`, or at least wrap them all in the same outer component that takes care of the functions that are essential to each (eg. `handleGoAwayClick`, `handleSignInClick`, `handleXClick`)?